### ISSUE TITLE
Dead-lock in AsyncProc

### ISSUE THREAD

**Issue by g4s8** [zerocracy/farm#1590]
After #1567 fix `AsyncProc` may catch a dead-lock. It processes few SQS messages and then stops. I've reverted commits for this PR to get Zerocrat working, it is c3091421a341a6eb9824385387329c14937aea5f and e744542d6492cf0924176d16481c7e12edc83c73. We need to reimplement this `AsyncProc` and unit-test it to be sure that it's working.

**Comment by g4s8** [comment]
@0crat assign @carlosmiranda

**Comment by carlosmiranda** [comment]
@g4s8 see PR #1597.

**Comment by 0crat** [comment]
@carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by carlosmiranda** [comment]
@0crat wait for merge of #1597

**Comment by 0crat** [comment]
> @0crat wait for merge of #1597 ([here](https://github.com/zerocracy/farm/issues/1590#issuecomment-414095209))

@carlosmiranda The impediment for [#1590](https://github.com/zerocracy/farm/issues/1590) was registered successfully by @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)

**Comment by g4s8** [comment]
@carlosmiranda thanks

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1590](https://github.com/zerocracy/farm/issues/1590) is now out of scope

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +35 point(s) just awarded to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)
